### PR TITLE
chore: update project structure and tooling

### LINKED ISSUE #133: Missing @typescript-eslint/parser dependency
### 🤔 What's the problem you've observed?

When running `npm install` and followed by `npm run eslint`, the later command fails as `@typescript-eslint/parser` is not installed.

```
Error: Failed to load parser '@typescript-eslint/parser' declared in '.eslintrc.json': Cannot find module '@typescript-eslint/parser'
```

### ✨ Do you have a proposal for making it better?

Include `@typescript-eslint/parser` in dev dependencies.

### 📚 Any additional context?

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/ISSUE_TEMPLATE/developer_experience.md)*

### PR DESCRIPTION
**Pull request by davidjgoss**
### 🤔 What's changed?

- Tests are now colocated with sources, not in a `test` dir
- Tests are named `*.spec.tsx`
- Tests are run with Jest, using jest-dom for assertions and testing library for rendering
- Test helpers are under a `test-utils` dir with a single entry point
- Mocha, sinon etc removed

Also fixes #133 while we're here.

### ⚡️ What's your motivation? 

Make this project more in keeping with React ecosystem conventions and thus more approachable for would-be external contributors. Also better DX generally with Jest (with React anyway).

### 🏷️ What kind of change is this?

- :bank: Refactoring/debt/DX (improvement to code design, tooling, documentation etc. without changing behaviour)

### ♻️ Anything particular you want feedback on?

Any major objections?

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
- [ ] I've changed the behaviour of the code
  - [ ] I have added/updated tests to cover my changes.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [ ] Users should know about my change
  - [ ] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../CHANGELOG.md), linking to this pull request.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

### CODE REVIEW

**Comment by aslakhellesoy** [comment]
Sounds good to me!

Jest has always felt bloated to me, but you’re probably right that it’s the testing tool of choice in the React world.

What kind of DX gets better with Jest in your opinion?

**Comment by davidjgoss** [comment]
> Jest has always felt bloated to me, but you’re probably right that it’s the testing tool of choice in the React world.

Exactly - for testing more "plain" JS I generally choose Mocha or [Ava](https://github.com/avajs/ava).

> What kind of DX gets better with Jest in your opinion?

- DOM stuff built in, doesn't have to be retrofitted
- `jest.fn()` and `jest.Mock` more ergonomic than sinon IMO (by virtue of being built in)
- Snapshot tests are useful _occasionally_
- Good assertions especially with `jest-dom` e.g. `expect(someElement).toHaveValue('foo')`

File: src/acceptance.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+describe('acceptance tests', () => {
+  jest.setTimeout(30000)
+
+  describe('with test data', () => {
+    const dir = __dirname + '/../testdata/good'
+    const files = fs.readdirSync(dir)
+
+    for (const file of files) {
+      if (file.match(/\.feature$/)) {
+        it(`can render ${file}`, async () => {
```
Comment (aurelien-reeves) [inline src/acceptance.spec.tsx:30]: I see no assertion in there.
I guess the test is failing if there is an error while rendering the component?

**Review by aurelien-reeves — APPROVED** [review]

File: src/acceptance.spec.tsx (hunk 1 above)
Comment (davidjgoss) [reply src/acceptance.spec.tsx:30]: > I guess the test is failing if there is an error while rendering the component?

Yeah I think that's the idea.

> Would it be possible to still have an assertion?

We could probably add a check to make sure the rendered element isn't empty, let me try that.

Comment (davidjgoss) [reply src/acceptance.spec.tsx:30]: Done!

**Pull request merged**